### DISCUSSION TITLE
Docsy as npm package inconsistency

### DISCUSSION THREAD

**Discussion by NatasaKralj** [google/docsy#1421] [Q&A] [answered]
Hello,

I'm trying to migrate a site from Docsy v0.1 (as a submodule) to Docsy v0.6.0 as an npm package. 

One issue I've run into is that installing Docsy via npm `npm install --save-dev google/docsy autoprefixer postcss-cli postcss` gets me the latest main build as a package, instead of the latest release version.

I noticed it due to the release version of 0.6.0 having Bootstrap pinned at 4.6.2, while the latest main has it at 5.2.3. 

I would love to be able to specify which release version of Docsy to install via npm, but there only seems to be the latest package available. Can we have more versions added to npm? Include at least the latest official release version along with the latest main build on npm?

I also noticed in npm info (`npm view google/docsy`) that the license information is marked as proprietary, isn't the Docsy license  Apache-2.0?

**Comment by deining** [comment]
> Hello,
> 
> I'm trying to migrate a site from Docsy v0.1 (as a submodule) to Docsy v0.6.0 as an npm package.
> 
> One issue I've run into is that installing Docsy via npm `npm install --save-dev google/docsy autoprefixer postcss-cli postcss` gets me the latest main build as a package, instead of the latest release version.
> 
> I noticed it due to the release version of 0.6.0 having Bootstrap pinned at 4.6.2, while the latest main has it at 5.2.3.
> 
> I would love to be able to specify which release version of Docsy to install via npm, but there only seems to be the latest package available.

Good point, indeed there is no npm package available for v0.6.0 .
As proof of concept / workaround, I created a [package](https://github.com/deining/docsy/pkgs/npm/docsy) in my docsy fork. You can use this one for now.
**CAVEAT:** this in an inofficial, one-time release, I won't provide any new packages in the future.

> Can we have more versions added to npm? Include at least the latest official release version along with the latest main build on npm?

I will open a issue for this shortly.

> I also noticed in npm info (`npm view google/docsy`) that the license information is marked as proprietary, isn't the Docsy license Apache-2.0?

Thanks for reporting, this is addressed with #1428.

**Reply by NatasaKralj** [reply]
Thanks for the reply @deining and the package. I happily await the addition of release versions to Docsy's npm packages.

**Reply by NatasaKralj** [reply]
@deining I'm having trouble using the package you created. The link in your reply 404's and I don't see any packages in your Docsy fork. Could you help a bit more? What am I missing?

**Comment by chalin** [comment, marked as answer]
@NatasaKralj, to get [v0.6.0 of Docsy](https://github.com/google/docsy/releases/tag/v0.6.0) (which is at hash ` 5597d43`), use the following command:

```console
$ npm install --save-dev google/docsy#semver:0.6.0
...
$ grep 'version.*docsy.git' package-lock.json
      "version": "git+ssh://git@github.com/google/docsy.git#5597d435dc74ce68240e0c3871addf24567493b0",
```

Let me know if that works for you. Thanks @deining for jumping in to help while I was busy with BSv5 work. I'll get to your PR soon, and open an issue for updating the docs.

> I would love to be able to specify which release version of Docsy to install via npm

You can use the syntax shown above to install any official release of Docsy.

@deining, I've merged your PR (and used it to bump up the version):

```console
$ npm view google/docsy       

docsy@0.7.0-dev | Apache-2.0 | deps: 2 | versions: 1
https://www.docsy.dev
...
```

**Reply by NatasaKralj** [reply]
Thank you @chalin this did the trick. Great that you also updated the documentation!

**Reply by chalin** [reply]
If possible, could you update the "answer" for this discussion? Thanks!

**Comment by chalin** [comment]
Doc update PR:

- #1439
